pkg/collector/reconcile/configmap.go

```diff
@@ -66,6 +66,10 @@ func desiredConfigMap(_ context.Context, params Params) corev1.ConfigMap {
 	name := naming.ConfigMap(params.Instance)
 	labels := collector.Labels(params.Instance)
 	labels["app.kubernetes.io/name"] = name
+	config, err := ReplaceConfig(params)
+	if err != nil {
+		params.Log.V(2).Info("failed to parse config: ", err)
```

Comment:
Is this always a failure to parse?  It looks like there are many possible errors this can return.

Reply:
Yes there can be multiple errors. Since the overall goal is to parse the config and replace it with the new `http_sd_config` entry, I thought the initial message can be "failed to parse config: " and then the err can talk about the actual error as to why the parsing failed.
But I think it may confuse the user. Does "failed to replace config: " make more sense? or should I return the err alone?

Reply:
Maybe a clue as to why we are replacing the config would be helpful: "failed to update prometheus config to use sharded targets"

Reply:
Sounds good! Thank you.